kernel/core/os.adb

```diff
@@ -300,22 +296,25 @@ is
                --  Already in the ready list
                exit;
             elsif os_get_task_priority (task_id) >
-               os_get_task_priority (index_id)
+              os_get_task_priority (index_id)
             then
-               prev_id := os_task_rw (index_id).prev;
-               os_task_rw (task_id).next  := index_id;
-               os_task_rw (index_id).prev := task_id;
-
-               if index_id = os_sched_get_current_list_head then
-                  os_task_rw (task_id).prev := OS_TASK_ID_NONE;
-                  os_sched_set_current_list_head (task_id);
-               else
-                  os_task_rw (task_id).prev := prev_id;
-                  if prev_id in os_task_id_param_t then
-                     os_task_rw (prev_id).next := task_id;
+               declare
+                  prev_id : constant os_task_id_t := os_task_rw (index_id).prev;
```

Comment:
Is it known to be better for proof to have local variables? I guess it might make sense.

Reply:
OK